SVA/lib/state.c

```diff
@@ -1284,6 +1288,11 @@ sva_init_stack (unsigned char * start_stackp,
     icontextp->valid |= IC_is_valid;
   }
 
+  if(vg && (oldThread->secmemSize) && (oldThread->flags & 1))
```

Comment:
This code should have a comment explaining what it is doing.  Also, the constant should be replaced with a const global variable that makes it clear what the constant represents.